src/main/java/lotto/controller/MainController.java

```diff
@@ -0,0 +1,130 @@
+package lotto.controller;
+
+import lotto.model.*;
+import lotto.util.LottoNumberCreator;
+import lotto.util.WinningRank;
+import lotto.view.InputView;
+import lotto.view.OutputView;
+
+import java.util.List;
+
+public class MainController {
+    private static LottoPrice lottoPrice;
+    private final InputView inputView;
+    private final OutputView outputView;
+
+    public MainController(InputView inputView, OutputView outputView) {
+        this.inputView = inputView;
+        this.outputView = outputView;
+    }
+
+    private static TotalLotto getTotalLotto(OutputView outputView, InputView inputView) {
+        outputView.printMessageForInputPurchaseAmount();
+        lottoPrice = inputView.getLottoPrice();
+        outputView.printNumberForPurchasedLottos(lottoPrice);
+        LottoNumberCreator lottoNumberCreator = new LottoNumberCreator(lottoPrice);
+        outputView.printPurchasedLottos(lottoNumberCreator.getTotalLotto());
+        return lottoNumberCreator.getTotalLotto();
+    }
+
+    private static Lotto getWinningNumberLotto(OutputView outputView, InputView inputView) {
+        outputView.printMessageForWinningNumbers();
+        WinningNumber winningNumber = inputView.getWinningNumbers();
+        return winningNumber.getLotto();
+    }
+
+    private static BonusNumber getBonusNumber(OutputView outputView, InputView inputView) {
+        outputView.printMessageForBonusNumber();
+        return inputView.getBonusNumber();
+    }
+
+    public void startLottoProgram() {
+        TotalLotto totalLotto = getTotalLotto(outputView, inputView);
+        Lotto winningNumberLotto = getWinningNumberLotto(outputView, inputView);
+        BonusNumber bonusNumber = getBonusNumber(outputView, inputView);
+        outputView.printMessageLotteryStatistics();
+        checkResult(totalLotto, winningNumberLotto, bonusNumber);
+        checkProfit(lottoPrice);
+
+    }
+
+    private void checkResult(TotalLotto totalLotto, Lotto winningNumberLotto, BonusNumber bonusNumber) {
+        for (Lotto lotto : totalLotto.getLottoList()) {
+            checkWhoIs1(lotto, winningNumberLotto);
+            checkWhoIs2(lotto, winningNumberLotto, bonusNumber);
+            checkWhoIs3(lotto, winningNumberLotto, bonusNumber);
+            checkWhoIs4(lotto, winningNumberLotto);
+            checkWhoIs5(lotto, winningNumberLotto);
+        }
+        outputView.printLotteryStatisticsFor_5(WinningRank.FIFTH.getCount());
+        outputView.printLotteryStatisticsFor_4(WinningRank.FOURTH.getCount());
+        outputView.printLotteryStatisticsFor_3(WinningRank.THIRD.getCount());
+        outputView.printLotteryStatisticsFor_2(WinningRank.SECOND.getCount());
+        outputView.printLotteryStatisticsFor_1(WinningRank.FIRST.getCount());
+    }
```

Comment:
출력에 관한 로직을 이곳에서 모두 하나하나 호출하기보단, 위 메서드를 private으로 만들고 public으로 한 메서드만 열어도 좋을 것 같습니다. 그리고 WinningRank에서는 파라미터에 따른 동작을 정의하도록 하면 호출부분이 훨씬 깔끔해질 것 같습니다
```suggestion
    private void checkResult(TotalLotto totalLotto, Lotto winningNumberLotto, BonusNumber bonusNumber) {
        for (Lotto lotto : totalLotto.getLottoList()) {
            determineRank(lotto, winningNumberLotto);
        }
        outputView.printLotteryStatistics(winningRank);
    }
```

Reply:
네네 좋은 의견 감사합니다~!